tools/mavlogdump.py

```diff
@@ -99,10 +100,10 @@
 if nottypes is not None:
     nottypes = nottypes.split(',')
 
-ext = os.path.splitext(filename)[1]
-isbin = ext in ['.bin', '.BIN', '.px4log']
-islog = ext in ['.log', '.LOG'] # NOTE: "islog" does not mean a tlog
-istlog = ext in ['.tlog', '.TLOG']
+mlog_type = str(type(mlog))
+isbin = "DFReader_binary" in mlog_type
+islog = "DFReader_text" in mlog_type # NOTE: "islog" does not mean a tlog
+istlog = "mavmmaplog" in mlog_type
```

Comment:
This should only ever be a dataflash log - if you look at the class you can see it's a DFLOG parser.

So something *very* weird is going on here :-)

Reply:
mavlogdump handles either DF binary, DF text or Mavlink/Tlog format files, and the class of the variable returned by `mlog = mavutil.mavlink_connection(...)` will reflect this.
I added a print on these flags during testing with all 3 file types (including the TLog named both as .log and .tlog), and the right flag gets set.